src/qibosoq/components.py

```diff
@@ -80,14 +81,33 @@ class Parameter(IntEnum):
     BIAS = auto()
     DURATION = auto()
 
+    @overload
+    @classmethod
+    def variants(cls, parameters: str) -> "Parameter":
+        """Convert a string to a Parameter."""
+
+    @overload
+    @classmethod
+    def variants(cls, parameters: Iterable[str]) -> Iterable["Parameter"]:
```

Comment:
Just saying `Iterable` it's a restricted version of the `Union`: you know that given an iterable, you'll obtain back another one, but it's not telling you that is an iterable *of the same kind*... but there is no better option 😞 

Unfortunately, when I wrote it I used a feature that is not a Python feature, even though there are actually people advocating for it
https://github.com/python/typing/issues/548

Reply:
Yes I saw it :-(